### PR TITLE
Add functions for tracking attendance in Class.java

### PR DESCRIPTION
**Pull request by seanpzk**
### Attendance tracking feature:
Add `startlesson` command. Test out by going through below steps:
- `add /c cs2103`
- `add /s John /c 1`
- `view /c 1`
- `startlesson /c cs2103` causes total lessons to increase by one
- `Student` under the `Class` will be marked absent while his/her attendance value remain unchanged

### Misc:
Add functions to create and update `totalLessons` of a `Class`
Add `MarkAllStudentAbsentCommand` and its `Parser` for future use
Add navigability from `Student` to `Class`
Created GUI for `StudentCard` to allow display of `Student` attendance 
Storage updated : `JsonAdaptedClass` now saves `totalLessons` of a Class

### CODE REVIEW

**Review by songgthu — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -158,6 +158,64 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+The relationship between variables can be summarised by this object diagram.
+
+<puml src="diagrams/RemoveStudentObjectDiagram.puml" alt="RemoveStudentObjectDiagram" />
+
+**Walkthrough**
+
+Step 1. `LogicManager` calls `removeStudentCommand#execute()` 
+
+Step 1. `removeStudentCommand` calls `Model#getClass(studentClassName)` to get `sClass`.
```
Comment (DonovanJJ) [inline docs/DeveloperGuide.md]: Update to DG looks neat and thorough 👍 
There seems to be a repetition of "Step 1", is this by design?

File: src/main/java/seedu/address/model/student/Student.java [hunk 2]
```diff
@@ -158,7 +187,7 @@ public int hashCode() {
     /**
      * Obtain String representation of the attendance of this Student.
      *
-     * @return String - Y - Present current class. N - Absent for current class
+     * @return String - Y - Present current class1. N - Absent for current class
```
Comment (DonovanJJ) [inline src/main/java/seedu/address/model/student/Student.java]: What would class1 mean here?

File: src/main/java/seedu/address/model/student/Student.java [hunk 3]
```diff
@@ -27,40 +27,68 @@ public class Student {
...
 
     /**
-     * If only name is provided.
+     * Constructs a Student with Name and Index that represents one-based class index.
+     *
+     * @param name Name to represent Student.
      */
     public Student(Name name) {
-        requireNonNull(name);
+        requireAllNonNull(name);
```
Comment (DonovanJJ) [inline src/main/java/seedu/address/model/student/Student.java]: Perhaps we could just stick with requireNonNull here since only one argument is passed

**Review by DonovanJJ — CHANGES_REQUESTED** [review]
LGTM 👍 
Just some nitpicking but overall really well documented and easy to follow!

**Review by DonovanJJ — APPROVED** [review]
LGTM!

**Pull request merged**